contrib/pylibpod/pylibpod/libs/containers.py

```diff
@@ -0,0 +1,207 @@
+"""Models for manipulating containers and storage."""
+import collections
+import functools
+import json
+
+
+class Container(collections.UserDict):
+    """Model for a container."""
+
+    def __init__(self, client, id, data):
+        """Construct Container Model."""
+        super(Container, self).__init__(data)
+
+        self._client = client
+        self._id = id
+
+        self._refresh(data)
+        assert self._id == self.data['id'],\
+            f'Requested container id({self._id}) does not match store id({self.id})'
+
+    def __getitem__(self, key):
+        """Get items from parent dict + apply aliases."""
+        if key == 'running':
+            key = 'containerrunning'
+        return super().__getitem__(key)
+
+    def _refresh(self, data):
+        super().update(data)
+        for k, v in data.items():
+            setattr(self, k, v)
+        setattr(self, 'running', data['containerrunning'])
+
+    def refresh(self):
+        """Refresh status fields for this container."""
+        ctnr = Containers(self._client).get(self.id)
+        self._refresh(ctnr)
+
+    def attach(self, detach_key=None, no_stdin=False, sig_proxy=True):
+        """Attach to running container."""
+        with self._client() as podman:
+            # TODO: streaming and port magic occur, need arguements
+            podman.AttachToContainer()
+
+    def processes(self):
+        """Show processes running in container."""
+        with self._client() as podman:
+            results = podman.ListContainerProcesses(self.id)
+        for p in results['container']:
+            yield p
+
+    def changes(self):
+        """Retrieve container changes."""
+        with self._client() as podman:
+            results = podman.ListContainerChanges(self.id)
```

Comment:
does it make sense to combine these two into a return? dealers choice

Reply:
@baude I've kept them separate for two reasons:

1) the client socket connection is open for the duration of the with statement.  So I've coded that to be the minimum number statements for the day when we go remote.  processes() could have yielded within the with, but then the connection to podman would be open until the caller exhausted the list of processes.
2) I personally don't like the style of method()[0]  unless the method call is trivial. :)